### PR TITLE
Improve RISC-V recursive method invocations

### PR DESCRIPTION
**Pull request by 0xdaryl**
* Replace recursive method checks with `isRecursiveMethodTarget` Compilation query.
* Provide System linkage implementations of entry point APIs
* Use entryPointFromCompiledMethod linkage API for RISC-V recursive calls

Signed-off-by: Daryl Maier <maier@ca.ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Just FYI @janvrany

**Comment by 0xdaryl** [comment]
The CI failure on macOS could not have been caused by this PR (which is RISC-V specific).  I've opened #5581 to track.

**Comment by 0xdaryl** [comment]
@aviansie-ben : would you mind reviewing/merging this simple change please?

**Comment by 0xdaryl** [comment]
@janvrany : FYI.  Would you mind building this PR to see if there are any problems?  The scope has grown a little bit since my first push.

**Comment by janvrany** [comment]
@0xdaryl : sure, will have a look tomorrow and let you know!

**Comment by janvrany** [comment]
@0xdaryl, sorry, got distracted. There are two little issues causing compilation to fail. With the below patch, 
the branch compile and passes `MinimalTests`, most importantly `RecursiveFibonacci`, so I'd say code is good. 

```
diff --git a/compiler/riscv/codegen/OMRCodeGenerator.hpp b/compiler/riscv/codegen/OMRCodeGenerator.hpp
index 0a5e0e77b..384ef157f 100644
--- a/compiler/riscv/codegen/OMRCodeGenerator.hpp
+++ b/compiler/riscv/codegen/OMRCodeGenerator.hpp
@@ -78,7 +78,7 @@ public:
 
    CodeGenerator();
 
-   initialize();
+   void initialize();
 
    /**
     * @brief AArch64 hook to begin instruction selection
diff --git a/compiler/riscv/codegen/RVInstruction.cpp b/compiler/riscv/codegen/RVInstruction.cpp
index 68b864019..27d3f4ba0 100644
--- a/compiler/riscv/codegen/RVInstruction.cpp
+++ b/compiler/riscv/codegen/RVInstruction.cpp
@@ -48,6 +48,7 @@
 #include "infra/Assert.hpp"                       // for TR_ASSERT
 #include "codegen/RVOutOfLineCodeSection.hpp"
 #include "codegen/GenerateInstructions.hpp"
+#include "codegen/Linkage.hpp"
 
 #define TR_RISCV_RTYPE(insn, rd, rs1, rs2) \
   ((insn) | ((rd) << OP_SH_RD) | ((rs1) << OP_SH_RS1) | ((rs2) << OP_SH_RS2))
```

Now, the missing `void` has actually been there since 4e028e2a43 (#5566)

**Comment by 0xdaryl** [comment]
Thanks @janvrany, and for catching my earlier bonehead compile error.  I've pushed the fixes.

**Comment by janvrany** [comment]
A polite ping. 
Is there anything blocking this from being merged? As of now, RISC-V codegen does not compile due to a slip in `OMRCodeGenerator.hpp` - this PR (commit f9a71c7), fixes the problem. 

Or shall I open separare PR just to fix that?

**Comment by 0xdaryl** [comment]
No, it's ready.  I just need a committer to take it.

**Review by aviansie-ben — APPROVED** [review]
LGTM

**Comment by aviansie-ben** [comment]
Since this change only touches the RISC-V codegen and we still don't have RISC-V CI, I'll go ahead and merge without further testing.

**Pull request merged**
